app/src/main/java/co/nimblehq/compose/crypto/ui/theme/Theme.kt

```diff
@@ -3,34 +3,46 @@ package co.nimblehq.compose.crypto.ui.theme
 
 import androidx.compose.foundation.isSystemInDarkTheme
 import androidx.compose.material.*
-import androidx.compose.runtime.Composable
-import co.nimblehq.compose.crypto.ui.theme.Color.DarkJungleGreen
-import co.nimblehq.compose.crypto.ui.theme.Color.Guyabano
-
-object Palette {
-    val ComposeLightPalette = lightColors(
-        surface = Guyabano,
-    )
-
-    val ComposeDarkPalette = darkColors(
-        surface = DarkJungleGreen
-    )
-}
+import androidx.compose.runtime.*
 
 @Composable
 fun ComposeTheme(
     darkTheme: Boolean = isSystemInDarkTheme(),
     content: @Composable () -> Unit
 ) {
     val colors = if (darkTheme) {
-        Palette.ComposeDarkPalette
+        DarkColorPalette
     } else {
-        Palette.ComposeLightPalette
+        LightColorPalette
+    }
+
+    CompositionLocalProvider(LocalColors provides colors) {
+        MaterialTheme(
+            colors = colors.themeColors,
+            shapes = Shape.ComposeShapes,
+            content = content
+        )
     }
+}
+
+/**
+ * Alternate to [MaterialTheme] allowing us to add our own theme systems
+ * or to extend [MaterialTheme]'s types e.g. return our own [Colors] extension.
+ */
+object AppTheme {
+
+    val colors: AppColors
+        @Composable
+        @ReadOnlyComposable
+        get() = LocalColors.current
+
+    val typography: Typography
```

Comment:
@luongvo May I know about your plan for handling `typography` here? Are you to suggest following `MaterialTheme` for text as well? If I'm not wrong, by following this, we might need to call text like this `AppTheme.typography.h1` or `AppTheme.typography.subtitle1` 🤔

Reply:
@Wadeewee I think for consistency, we will try to make AppTheme reflects what we need in MaterialTheme, so in this case, we will call `AppTheme.typography.h1` or `AppTheme.typography.subtitle1`.

Reply:
@luongvo I'm fine on both ways, but following the `MaterialTheme` style, I think we can't apply the `color` for `textStyle` like the previous implementation, e.g., `lightSilverMedium16()` but it's not a biggie for the color of text we can directly set the color in `Text`.

Reply:
But I have one more concern about the default style if we follow `MaterialTheme` for the text. Let's say we want `subtitle1` that's `fontSize = 16.sp`, `letterSpacing = 0.15.sp`, and `fontWeight = FontWeight.Normal` by default. The concern is if we want another one for `Bold`  that's  `fontWeight = FontWeight.Bold`. How should we do about this 🙏   or Should we do `AppTheme.typography.subtitle1.copy(fontWeight = FontWeight.Bold)` 🤔

Reply:
@luongvo Agree with Pooh. I am not a fan of inheriting the MaterialTheme since it might cause unexpected configuration from the theme, it could force override our external config 🤔

Reply:
@Wadeewee Yes, we can
- define a new style in AppStyles `AppTheme.typography.subtitle1Bold`.
- or clone the new style directly like `AppTheme.typography.subtitle1.copy(fontWeight = FontWeight.Bold)`.

Defining a new style or cloning it directly in use is up to you. Even relying on MaterialTheme or not, depends on the design and app look & feel too. I don't see any issue here 💭 Could you guys please point out which's wrong or what is your suggestion? 👋  cc @hoangnguyen92dn

Reply:
> Could you guys please point out which's wrong or what is your suggestion? 

@luongvo For currently, We won't see the issue as the `jetpack-compose-crypto` is a small project, but TBH, I feel like it might be busy or messy when we work on the real project because in the real case, we will see many styles of text. IMO, I feel like the current way is more flexible than following the `MaterialTheme` cc @hoangnguyen92dn

Reply:
> but TBH, I feel like it might be busy or messy when we work on the real project because in the real case, we will see many styles of text.

@Wadeewee Your concern is valid. However, it depends on how can we manage them well. I just want to provide a common way for us to make customization, the next should belong to those who work on the project which extends our Compose Template (after contributing this back) 😄 

FYI: we have the same issue with Flutter https://github.com/nimblehq/varun-flutter/blob/c2229cba27d757487299a9254327dd8f506d7003/lib/views/job_detail/job_detail_page.dart#L196-L203